server/src/main/java/org/apache/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -118,54 +111,29 @@ public Map<String, List<VersionedSupervisorSpec>> getAll()
   {
     return ImmutableMap.copyOf(
         dbi.withHandle(
-            new HandleCallback<Map<String, List<VersionedSupervisorSpec>>>()
-            {
-              @Override
-              public Map<String, List<VersionedSupervisorSpec>> withHandle(Handle handle)
-              {
-                return handle.createQuery(
-                    StringUtils.format(
-                        "SELECT id, spec_id, created_date, payload FROM %1$s ORDER BY id DESC",
-                        getSupervisorsTable()
-                    )
-                ).map(
-                    new ResultSetMapper<Pair<String, VersionedSupervisorSpec>>()
-                    {
-                      @Override
-                      public Pair<String, VersionedSupervisorSpec> map(int index, ResultSet r, StatementContext ctx)
-                          throws SQLException
-                      {
-                        return Pair.of(
-                            r.getString("spec_id"),
-                            createVersionSupervisorSpecFromResponse(r)
-                        );
-                      }
-                    }
-                ).fold(
-                    new HashMap<>(),
-                    new Folder3<Map<String, List<VersionedSupervisorSpec>>, Pair<String, VersionedSupervisorSpec>>()
-                    {
-                      @Override
-                      public Map<String, List<VersionedSupervisorSpec>> fold(
-                          Map<String, List<VersionedSupervisorSpec>> retVal,
-                          Pair<String, VersionedSupervisorSpec> pair,
-                          FoldController foldController,
-                          StatementContext statementContext
-                      )
-                      {
-                        try {
-                          String specId = pair.lhs;
-                          retVal.computeIfAbsent(specId, sId -> new ArrayList<>()).add(pair.rhs);
-                          return retVal;
-                        }
-                        catch (Exception e) {
-                          throw new RuntimeException(e);
-                        }
-                      }
-                    }
-                );
-              }
-            }
+            (HandleCallback<Map<String, List<VersionedSupervisorSpec>>>) handle -> handle.createQuery(
```

Comment:
`handle` need not be cast here or in the other lambdas below.

Reply:
Without the cast in the `createQuery` case, the `fold` semantics will change a bit. So I left it as-is.